stdlib/distutils/archive_util.pyi

```diff
@@ -1,9 +1,22 @@
-from typing import Literal
+from _typeshed import FileDescriptorOrPath, StrPath
+from typing import Literal, overload
 
+@overload
```

Comment:
Why do we need these overloads? Looks like only `base_name` and `root_dir` are different, and when I look at the implementation in 3.11, I don't see any interdependency between those.

Reply:
If `root_dir` is not `None`, `base_name` can be `PathLike[str]`
https://github.com/python/cpython/blob/v3.11.9/Lib/distutils/archive_util.py#L225-L227

Reply:
I see. I'm not enthusiastic about adding overloads for cases like this where some types work by accident depending on the path taken through the body of the function.